client/src/components/Modal/tests/Modal-test.js

```diff
@@ -0,0 +1,106 @@
+/* global jest, test, describe, it, expect */
+
+import React from 'react';
+import { fireEvent, render, screen } from '@testing-library/react';
+import Modal from '../Modal';
+
+/** Component that printouts all of its props */
+const PrintProps = (componentName) => ({ children, ...propsToPrint }) => (
+  <div>
+    <div>{componentName}</div>
+    <ul>
+      {Object
+        .entries(propsToPrint)
+        .map(([key, value]) => (<li key={key}>{key}: {value.toString()}</li>))
+      }
+    </ul>
+    {children}
+  </div>
+);
+
+const makeProps = () => ({
+  isOpen: true,
+  className: 'my-modal',
+  modalClassName: 'my-modal-dialog',
+  size: 'sm',
+  onClosed: () => {},
+  title: 'Hello World!',
+  showCloseButton: true,
+});
+
+test('Modal renders', () => {
+  const root = render(
+    <Modal {...makeProps()}>My Content</Modal>
+  );
+
+  const modal = root.getByRole('dialog');
+  expect(modal.style.getPropertyValue('display')).toBe('block');
+  expect(modal.classList).toContain('my-modal-dialog');
+  expect(modal.classList).toContain('modal');
+
+  const modalContainer = root.getByRole('document');
+  expect(modalContainer.classList).toContain('modal-sm');
+
+  const header = root.getByRole('heading');
+  expect(header.textContent).toContain('Hello World!');
+
+  const modalBody = root.getByText('My Content');
+  expect(modalBody).toBeTruthy();
+
+  const closeButton = root.getByLabelText('Close');
+  expect(closeButton).toBeTruthy();
+});
+
+test('Modal with HTML title', () => {
```

Comment:
This test was added to cover the scenario where an object is passed to the title.